reference/src/layout/enums.md

```diff
@@ -351,90 +351,72 @@ enum Enum1<T> {
 }
 ```
 
-## Unresolved questions
-
 ### Layout of single variant enums
 
-[Issue #79.](https://github.com/rust-rfcs/unsafe-code-guidelines/issues/79)
-
-Enums that contain a **single variant** and which do not have an
-explicit `#[repr]` annotation are an important special case. Since
-there is only a single variant, the enum must be instantiated with
-that variant, which means that the enum is in fact equivalent to a
-struct. The question then is to what extent we should **guarantee**
-that the two share an equivalent layout, and also how to define the
-interaction with uninhabited types.
+**Single variant data-carrying*** enums without a `repr()` annotation have the
+same layout as the variant field. **Single variant fieldless** enums have the
+same layout as a unit struct.
 
-As presently implemented, the compiler will use the same layout for
-structs and for single variant enums (as long as they do not have a
-`#[repr]` annotation that overrides that choice). So, for example, the
-struct `SomeStruct` and the enum `SomeEnum` would have an equivalent
-layout ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3697ac684c3d021892694956df957653)):
+Here:
 
 ```rust
-struct SomeStruct;
-enum SomeEnum {
-  SomeVariant,
-}
-
+# use std::mem::{size_of, align_of};
+struct UnitStruct;
+enum SingleVariantFieldless { FieldlessVariant }
 # fn main() {
-# use std::mem;
-let x = SomeStruct;
-let y = SomeEnum::SomeVariant;
-assert_eq!(
-    mem::size_of_val(&x), 
-    mem::size_of_val(&y),
-    "types have different sizes",
-);
-println!("types both have size {}", std::mem::size_of_val(&x));
+assert_eq!(size_of::<SingleVariantFieldless>(), size_of::<UnitStruct>());
+assert_eq!(align_of::<SingleVariantFieldless>(), align_of::<UnitStruct>());
+// assert_eq!(call_abi_of::<SingleVariantFieldless>(), call_abi_of::<UnitStruct>());
+// assert_eq!(niches_of::<SingleVariantFieldless>(), niches_of::<UnitStruct>());
```

Comment:
What is the point of these commented-out assertions?

Reply:
We don't have any way to check those aspects of layout in code, so they are pseudo-code for that. I can remove them, since these are already implied when the text says "have the same layout". I thought the assertions made it clearer that by layout we meant _layout_.

Reply:
They're good asserts, and hopefully some day they'll be real rust functions as well

Reply:
The fact that they are commented out could also be interpreted as "this does not currently hold", as one would do in a test suite.

So, it should be made clearer that they are commented out just because the methods don't exist. Probably stating "these things are the same" as Rust asserts is not the clearest way anyway, and it *is* very verbose.